pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go

```diff
@@ -94,11 +94,7 @@ func (pl *DefaultPreemption) PostFilter(ctx context.Context, state *framework.Cy
 		Interface:  pl,
 	}
 
-	result, status := pe.Preempt(ctx, pod, m)
-	if status.Message() != "" {
-		return result, framework.NewStatus(status.Code(), "preemption: "+status.Message())
-	}
-	return result, status
+	return pe.Preempt(ctx, pod, m)
```

Comment:
We are missing some context here (the `preemption: ` string). But also, `Preempt` is already parsing a `FitError`.

Can you avoid losing this context?

Reply:
My feeling is that it's not worth adding the `preemption: ` prefix by additional decompose/recompose efforts. By reading the concrete status message and its context, it should be self-descriptive to tell it's a PostFilter message. @chendave any thoughts?

Reply:
(somehow my previous comment was gone...)

I _thought_ the `preemption: ` prefix was not that necessary as it can be inferred by the context. But by rethinking it a bit, I'm fine with keeping it, but I don't think it makes sense to add the prefix to a particular plugin. Some ideas below:

1. We add it to PostFilter plugin
2. "preemption: " sounds a bit exclusive, how about "PostFilter:"
3. Another idea: the current format for preemption/PostFilter has two colons, which doesn't read well:
    ```
    I0706 11:16:12.539261   58361 scheduler.go:360] "Unable to schedule pod; no fit; waiting" pod="default/pause-req" err="0/10 nodes are available: 10 Insufficient cpu. preemption: 0/10 nodes are available: 10 No preemption victims found for incoming pod."
    ```
    How about changing it `[PostFilter]`:  (please check latest commit)
    ```
    I0706 11:16:12.539261   58361 scheduler.go:360] "Unable to schedule pod; no fit; waiting" pod="default/pause-req" err="0/10 nodes are available: 10 Insufficient cpu. [PostFilter] 0/10 nodes are available: 10 No preemption victims found for incoming pod."
    ```
    (Maybe we can also add a `[Filter]` prefix for filter message)

Reply:
I would not add `[Filter]` because it would increase the length of the event message unnecessarily.

The problem with `PostFilter` is that it's spilling internal details into user-visible messages. That's why I suggested "preemption" (probably better to say "Preemption"), but then it needs to be specific to the plugin. Now, In the examples you show, preemption can be deduced. But if the error message is about other scheduling directive (tolerations, etc.), it's not obvious.

Any thoughts about `Preempt` already parsing `FitError`?

Reply:
If we lean on the "preemption: " prefix, we have to decompose/re-compose the FitErrror. Please check the latest commit.

But it may not be that concerning as it's like we evaluate the error in the middle (`DefaultPreemption.PostFilter()`), which stored the msgs in `.reasons`, and that can be reused afterwards in https://github.com/kubernetes/kubernetes/blob/65361245edc10fea7ef9d30f03d0de24b8939da8/pkg/scheduler/schedule_one.go#L112